src/msg/async/AsyncMessenger.cc

```diff
@@ -291,7 +291,15 @@ void AsyncMessenger::ready()
 {
   ldout(cct,10) << __func__ << " " << get_myaddr() << dendl;
 
-  stack->start();
+  stack->ready();
+  if (pending_bind) {
```

Comment:
I thought so too, but the logic to find a free port to listen on is below the RDMAStack.